### PR TITLE
automake117x: init 1.17

### PR DESCRIPTION
**Pull request by trofi**
Switch to it by default will happen separately.

Changes: https://lists.gnu.org/archive/html/info-gnu/2024-07/msg00006.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/misc/automake/automake-1.17.x.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/automake/automake-1.17.x.nix:1]: Shouldn't we go the approach of a generic.nix file here isntead of duplicating code?

Comment (trofi) [reply pkgs/development/tools/misc/automake/automake-1.17.x.nix:1]: What would be a good example of a by-name package with `generic.nix` pattern?

Comment (Atemu) [reply pkgs/development/tools/misc/automake/automake-1.17.x.nix:1]: I don't know of an established pattern here but I think you could implement the generic.nix pattern as usual (inherit from the callPackage in all-packages) but simply use the by-name path.

Comment (Atemu) [reply pkgs/development/tools/misc/automake/automake-1.17.x.nix:1]: I do think we can make an exception here though as we're merely following the established pattern for versions of this package.

**Review by Atemu — APPROVED** [review]
I'd prefer if the generic.nix pattern was used so that we don't need to duplicate code here but that isn't the status quo.

File: pkgs/development/tools/misc/automake/automake-1.17.x.nix
Comment (trofi) [reply pkgs/development/tools/misc/automake/automake-1.17.x.nix:1]: > inherit from the callPackage in all-packages

Ah, I see https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#recommendation-for-new-packages-with-multiple-versions suggests a few workarounds. I can try that one instead.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**